### DISCUSSION TITLE
Some queries regarding Envoy Gateway

### DISCUSSION THREAD

**Discussion by saurabh-tiket** [envoyproxy/gateway#1533] [General]
I tried to install the Envoy gateway using https://gateway.envoyproxy.io/v0.4.0/user/quickstart.html

First is to Install the Gateway API CRDs and Envoy Gateway

Then we install the GatewayClass, Gateway, HTTPRoute and example app.

After installation following are my resources 

```
 $ k get all -n envoy-gateway-system                                    
NAME                                              READY   STATUS             RESTARTS         AGE
pod/envoy-gateway-55fb4b5bf-2lhjp                 2/2     Running            0                8d
pod/envoy-gateway-55fb4b5bf-hnpkt                 1/2     CrashLoopBackOff   1785 (53s ago)   6d19h
pod/envoy-default-eg-696e6672-59b5f8c9c9-nspr4   1/1     Running            0                5h32m

NAME                                    TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)               AGE
service/envoy-gateway                   ClusterIP      10.21.213.111   <none>           18000/TCP,18001/TCP   8d
service/envoy-gateway-metrics-service   ClusterIP      10.21.207.85    <none>           8443/TCP              8d
service/envoy-default-eg-696e6672      LoadBalancer   10.21.162.162   X.X.X.X          80:30533/TCP          6d2h
service/new-eg-neg                      ClusterIP      10.21.68.30     <none>           80/TCP                7d18h

NAME                                         READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/envoy-gateway                1/2     2            1           8d
deployment.apps/envoy-default-eg-696e6672   1/1     1            1           6d2h

NAME                                                    DESIRED   CURRENT   READY   AGE
replicaset.apps/envoy-gateway-55fb4b5bf                 2         2         1       8d
replicaset.apps/envoy-infra-ns-eg-696e6672-59b5f8c9c9   1         1         1       6d2h
```

I wanted to understand the difference between deployment  `deployment.apps/envoy-gateway` which is installed via helm and `deployment.apps/envoy-default-eg-696e6672` which is installed by kubectl 

Pardon my lack of knowledge regarding Envoy proxy in general. Can someone help me understand the difference `Envoy Gateway` and `Gateway` 

Also, in case of traffic increase should we scale up `Envoy Gateway` and `Gateway` both? 

Any link or videos explaining the concepts would be really helpful 

CC: @arkodg

**Comment by arkodg** [comment]
great questions @saurabh-tiket 
Looks like there is room for improvement in the Quickstart docs to make things more clear.
* `deployment.apps/envoy-default-eg-696e6672` is the [EnvoyProxy ](https://www.envoyproxy.io)instance that is serving the traffic (data plane) and 
`deployment.apps/envoy-gateway` is the EnvoyGateway instance that is managing & programing the data plane (creating the EnvoyProxy deployment as well as pushing networking/security config into it)
* Increase in traffic would mean more resources of Envoy Proxy being utilized which may need resource scaling
* There are user guides available here https://gateway.envoyproxy.io/v0.4.0/user_docs.html and links to past presentations available here https://gateway.envoyproxy.io/v0.4.0/presentations.html

**Reply by saurabh-tiket** [reply]
Thanks for the clarification. In case of `EnvoyProxy` Pods increase do we also need to vertically or horizontally scale the `EnvoyGateway`  Pod

Also, is there any possibility of adding HPA for `EnvoyProxy` service? 

@arkodg

**Reply by arkodg** [reply]
horizontal or vertical scaling depends based on type of traffic and type of operations performed on those traffic flows, unfortinately cannot be generalized, but the goal would be to come up with sane `resource` defaults so horizontal scaling should solve the performance bottleneck in majority of the cases.

HPA should be possible, needs enhancement in the docs to showcase the feature https://github.com/envoyproxy/gateway/issues/703
